### PR TITLE
(INT-1914) : Update for Public API Documentation

### PR DESCRIPTION
**Pull request by fahimmahmoodmir**
- Added section for created a dedicated user for Public API and Diagram

### CODE REVIEW

**Comment by fahimmahmoodmir** [comment]
Updated Section

image

**Comment by snikic** [comment]
Ah I don´t like the diagram too much, the diagram does not show the distinction between the agent credentials used for the bot and the one used for the public API (only when you read the labels)
Can you share the link to the diagram so I can propose something on it?+

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md [hunk 1]
```diff
@@ -39,21 +39,30 @@ To identify the Third-Party Bots API endpoint user, first get domain information
...
+
+Figure 2.1 shows a simple scenario below where a dedicated Public API User is first making a login call and then sending "Done" message to and ongoing conversation of a Customer with a Third-Party Bot.
+
+<img class="fancyimage" style="width:900px" src="img/tpbPublicApi/usage-diagram-public-api-message.png">
+Figure 2.1 Public API simple flow of sending messages command
+
+#### Step 3. Get Bearer Token
+
+To use our Public API you must perform a login request to Third-Party Bots API domain which will in response send you a Third-Party Bots bearer token. Moreover, You do not need to perform Login Request multiple times. One successful login call will send a bearer that can be used with multiple commands of Public API. There are two ways in which you can perform the login.
```
Comment (ahuppert) [inline pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: Hard to comment here on a specific part. I always find it useful to use a sensible max line length in the markdown as well even if it's not enforce it the project. Positive side effect would be that i can comment easier on what I think should be changed.

"Moreover, You do not need to perform Login Request multiple times"
"Moreover, you do not need to perform login request multiple times"

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md [hunk 2]
```diff
@@ -39,21 +39,30 @@ To identify the Third-Party Bots API endpoint user, first get domain information
 | z2.bot-connectors.liveperson.net | https://bot-platform-api.emea.fs.liveperson.com/ |
 | z3.bot-connectors.liveperson.net | https://bot-platform-api.apac.fs.liveperson.com/ |
 
-#### Step 2. Get Bearer Token
+#### Step 2. Create a Public API Bot User
 
-To use our Public API you must perform a login request to Third-Party Bots API domain which will in response send you a Third-Party Bots bearer token. There are two ways in which you can perform the login.
+In order to use Public API you must create a dedicated Public API bot agent/user via LivePerson User Management UI. You must not use this dedicated Public API agent/user user in any other LivePerson business process. For example, Using the same Public API Bot Agent to create a Third-Party Bot or using this Bot Agent user to call other LivePerson APIs. The reason for this requirement is to ensure Public API bot agent's session is not invalidated by any other service/business process.
```
Comment (ahuppert) [inline pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: For example, using the same Public API bot agent

Comment (ahuppert) [reply pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: using this bot agent

Comment (ahuppert) [reply pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: The reason for this requirement is to ensure that the session created by the Public API is not

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md [hunk 3]
```diff
@@ -39,21 +39,30 @@ To identify the Third-Party Bots API endpoint user, first get domain information
...
+
+To use our Public API you must perform a login request to Third-Party Bots API domain which will in response send you a Third-Party Bots bearer token. Moreover, You do not need to perform Login Request multiple times. One successful login call will send a bearer that can be used with multiple commands of Public API. There are two ways in which you can perform the login.
 
 <ul>
   <li>Bearer Token via Username/Password</li>
   <li>Bearer Token via API (OAuth)</li>
 </ul>
 
-To perform login requests you will need a valid Bot user created via LivePerson User Management UI. User must be `Enabled`, have minimum `Agent` role and have a Login Method selected either to `Password` or `API Key` as shown in Figure 2.1.
+To perform login requests you will need a valid dedicated Public API Bot user created via LivePerson User Management UI. User must be `Enabled`, have minimum `Agent` role and have a Login Method selected either to `Password` or `API Key` as shown in Figure 3.1.
```
Comment (ahuppert) [inline pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: dedicated Public API bot user

Comment (ahuppert) [reply pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: have the login method set either to

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md [hunk 4]
```diff
@@ -347,7 +356,7 @@ Example payload of the request with Simple Text, Pause/Delay, Private Text, [Str
 **Example cURL**:
 
 {: .important}
-**Please note** Make sure to replace [`{botDomain}`](#step-1-identify-the-third-party-bots-api-domain), `{accountId}`, `{conversationId}` and [`{bearerToken}`](#step-2-get-bearer-token) from the below command with your information
```
Comment (ahuppert) [inline pages/Documents/ConversationalAI/Third-PartyBots/public-api.md:350]: I think it's unnecessary to have the "please note" remark if it's already visualised with {: .important}
In general we should keep a consistent style for this throughout the third party bots docs.
What do you think? Check also you changes below

**Review by ahuppert — COMMENTED** [review]
Some comments on things i noticed.

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md (hunk 4 above)
Comment (fahimmahmoodmir) [reply pages/Documents/ConversationalAI/Third-PartyBots/public-api.md:350]: Sure removed all please notes in the important part in this file

**Comment by fahimmahmoodmir** [comment]
image
The New Photo i have added which @snikic  created

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md [hunk 5]
```diff
@@ -41,23 +41,23 @@ To identify the Third-Party Bots API endpoint user, first get domain information
...
+Figure 2.1 shows a simple scenario below where a dedicated Public API user is first making a login call and then sending "Done" message to and ongoing conversation of a Customer with a Third-Party Bot.
 
 <img class="fancyimage" style="width:900px" src="img/tpbPublicApi/usage-diagram-public-api-message.png">
 Figure 2.1 Public API simple flow of sending messages command
 
 #### Step 3. Get Bearer Token
 
-To use our Public API you must perform a login request to Third-Party Bots API domain which will in response send you a Third-Party Bots bearer token. Moreover, You do not need to perform Login Request multiple times. One successful login call will send a bearer that can be used with multiple commands of Public API. There are two ways in which you can perform the login.
+To use our Public API you must perform a login request to Third-Party Bots API domain which will in response send you a Third-Party Bots bearer token. Moreover, you do not need to perform Login Request multiple times. One successful login call will send a bearer that can be used with multiple commands of Public API. There are two ways in which you can perform the login.
```
Comment (ahuppert) [inline pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: Moreover, you do not need to perform **the login request** multiple times.

File: pages/Documents/ConversationalAI/Third-PartyBots/public-api.md [hunk 6]
```diff
@@ -41,23 +41,23 @@ To identify the Third-Party Bots API endpoint user, first get domain information
...
 
 <ul>
   <li>Bearer Token via Username/Password</li>
   <li>Bearer Token via API (OAuth)</li>
 </ul>
 
-To perform login requests you will need a valid dedicated Public API Bot user created via LivePerson User Management UI. User must be `Enabled`, have minimum `Agent` role and have a Login Method selected either to `Password` or `API Key` as shown in Figure 3.1.
+To perform login requests you will need a valid dedicated Public API bot user created via LivePerson User Management UI. User must be `Enabled`, have minimum `Agent` role and have the login method set either to `Password` or `API Key` as shown in Figure 3.1.
```
Comment (ahuppert) [inline pages/Documents/ConversationalAI/Third-PartyBots/public-api.md]: valid and dedicated

**Review by ahuppert — COMMENTED** [review]
Some small remarks on the changes you have done.
Also what do you think about the line break suggestion?

**Comment by ahuppert** [comment]
approved (can't add another review)

**Pull request merged**